Userland/Libraries/LibWeb/Painting/PaintableBox.h

```diff
@@ -25,7 +25,7 @@ class PaintableBox : public Paintable {
 
     virtual void paint(PaintContext&, PaintPhase) const override;
 
-    bool is_visible() const { return layout_box().is_visible(); }
+    virtual bool is_visible(Optional<PaintContext const&> = {}) const { return layout_box().is_visible(); }
```

Comment:
Not sure why this function needs to have two modes, can we just always pass `PaintContext const&` to keep it simple? :)

Reply:
Previously, it needed to be optional as `is_visible()` was used from hit testing too. I've now removed this change and instead added `Paintable::forms_unconnected_subtree()`, which is true for SVG masks. That means SVG masks won't be connected to the (main) paint tree, so can't be painted (unless referenced by another element's `mask` attribute). I think that's a little nicer of a solution.

Reply:
That does seem nicer!